svgnative/src/SVGDocumentImpl.cpp

```diff
@@ -1070,6 +1070,24 @@ void SVGDocumentImpl::TraverseTree(const ColorMap& colorMap, const Element& elem
         {
             ApplyCSSStyle(reference.classNames, graphicStyle, fillStyle, strokeStyle);
             auto saveRestore = SaveRestoreHelper{mRenderer, reference.graphicStyle};
+            if ((*(refIt->second)).Type() == ElementType::kGraphic)
+            {
+                Graphic& graphic = static_cast<Graphic&>(*(refIt->second));
+                // it will call assignment operator and pass fillStyle from reference to graphic
+                graphic = reference;
+            }
+            else if((*(refIt->second)).Type() == ElementType::kGroup)
```

Comment:
Thanks Dirk, i get your point of multiple nested groups if given, and i modified changes recently again to make code more generalised.

Reply:
<g fill=“yellow”>
    <rect id=“r” width=“100” height=“100"/>
</g>
<use xlink:href=“#r” x=“120" />

in above example ,with the original code (not including any recent changes) the output is first rect is yellow and second is also yellow. even our expectation is that the first rect should be yellow and the second is black.

we have a stack of fill and stroke styles. the first rect takes fill style from parent . That FillStyle will be passed as a reference to function “ParseGraphicProperties” as it modifies FillStyle if any fill attribute found.currently in this scenario, it did not able to modify because no fill attribute is present. Then “Graphic” element with its FillStyle is pushed in stack. **There is no way to distinguish its parent FillStyle and current one.**
when reference element is pointing to any graphic element, simply graphic element will be drawn as it was previously stored in stack. so “yellow” rect will be drawn. if there is any styling applied to reference object it will simply ignored.

i have also debug the area you pointed in discussion
 auto saveRestore = SaveRestoreHelper{mRenderer, reference.graphicStyle};

it simple saves the graphic style , but fill and stroke properties are not stored anywhere which actually fills the rect.